scylla/src/network/connection_pool.rs

```diff
@@ -934,6 +943,17 @@ impl PoolRefiller {
         // As this may may involve resolving a hostname, the whole operation is async.
         let endpoint_fut = self.maybe_translate_for_serverless(endpoint);
 
+        let count_in_metrics = {
+            let metrics = self.metrics.clone();
+            move |connect_result: &Result<_, ConnectionError>| {
+                if connect_result.is_ok() {
```

Comment:
Nit: prefer `Arc::clone` to `.clone()` to make it explicit that you are not cloning the underlying data structure.